app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx

```diff
@@ -23,33 +28,53 @@ const { NOT_CONNECTED, OPEN_APP, CONNECTED } = LEDGER_CONNECTION_STAGES
 
 type LedgerConnectionStage = $Values<typeof LEDGER_CONNECTION_STAGES>
 
+type LedgerPublicKey = { account: number, key: string }
+
 type Props = {
   progress: string,
   login: Function,
   connect: Function,
-  publicKey: ?string,
   error: ?string
 }
 
+type SelectOption = {
+  label: string,
+  value: string
+}
+
 type State = {
   ledgerStage: LedgerConnectionStage,
-  isLoading: boolean
+  isLoading: boolean,
+  address: SelectOption | null,
+  publicKeys: Array<LedgerPublicKey>,
+  loadingPublicKeys: boolean
 }
 
 const POLL_FREQUENCY_MS = 1000
 
 export default class LoginLedgerNanoS extends React.Component<Props, State> {
   constructor(props: Props) {
     super(props)
-
-    this.state = this.computeStateFromProps(props)
+    this.state = {
+      ...this.computeStateFromProps(props),
+      address: null,
```

Comment:
I think this is a bit confusing.. usually, an `address` is a string that represents the public address.. but here its actually an option. How about renaming it to `addressOption`?